sklearn/metrics/tests/test_classification.py

```diff
@@ -2530,6 +2530,15 @@ def test_log_loss():
     loss = log_loss(y_true, y_score2, labels=[1, 2, 3])
     assert_almost_equal(loss, 1.0630345, decimal=6)
 
+    # ensure np.float32 and np.float16 inputs give correct output
```

Comment:
Could you isolate this part in a single test such that we can document the issue:

```python
def test_log_loss_eps_auto():
    """Check the behaviour of `eps="auto"` that changes depending on the input
    array dtype.
    Non-regression test for:
    https://github.com/scikit-learn/scikit-learn/issues/24315
    """
    ...